python/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+# NOTE: These must be checked before including Cython packages to ensure we are using the right python version.
+# Segfaults could happen if we are using the wrong version.
+set(PYTHON_INCLUDE_DIR ${PYTHON_INCLUDE_DIRS})
+set(PYTHON_LIBRARY ${PYTHON_LIBRARIES})
+
+include(FindCython)
```

Comment:
`FindCython` runs `find_package( PythonInterp )`. However, in `setup.py`, when we call `build.sh`, we pass in the Python executable via

```python
subprocess.check_call(["../build.sh", "-p", sys.executable])
```

The Python we want to find is the one that is used to call `setup.py`. So, in the scenario where the Python used to call `setup.py` is different from the first Python on the user's PATH, will we find the right Cython executable?

Reply:
Good point, let me see if we can pass this through somehow

Reply:
done